apis/v1alpha2/validation/grpcroute.go

```diff
@@ -63,13 +67,38 @@ func validateGRPCRouteRules(rules []gatewayv1a2.GRPCRouteRule, path *field.Path)
 	return errs
 }
 
-// validateRuleMatches validates that at least one of the fields Service or Method of
-// GRPCMethodMatch to be specified
+// validateRuleMatches validates GRPCMethodMatch
 func validateRuleMatches(matches []gatewayv1a2.GRPCRouteMatch, path *field.Path) field.ErrorList {
 	var errs field.ErrorList
 	for i, m := range matches {
-		if m.Method != nil && m.Method.Service == nil && m.Method.Method == nil {
-			errs = append(errs, field.Required(path.Index(i).Child("method"), "one or both of `service` or `method` must be specified"))
+		if m.Method != nil {
+			if m.Method.Service == nil && m.Method.Method == nil {
+				errs = append(errs, field.Required(path.Index(i).Child("method"), "one or both of `service` or `method` must be specified"))
+			}
+			// GRPCRoute method matcher admits two types: Exact and RegularExpression.
+			// If not specified, the match will be treated as type Exact
+			if m.Method.Type == nil || *m.Method.Type == gatewayv1a2.GRPCMethodMatchExact {
+				if m.Method.Service != nil {
+					r, err := regexp.Compile(validServiceName)
```

Comment:
instead of compiling the regex with every validation we can make it faster by compiling them as a global var.

Reply:
also will help in not having to assert this error because this error is a design time error (as in if the expression is bad it will aways be an error) if it is good then it will always compile (unless you are running out of memory, which highly unlikely because it is a global var and is set before anything else).

Reply:
Fixed in #1972.